src/lspclient.rs

```diff
@@ -0,0 +1,126 @@
+use std::str::FromStr;
+
+use bitcoin::secp256k1::PublicKey;
+use reqwest::Client;
+use serde::{Deserialize, Serialize};
+
+pub(crate) struct LspClient {
+    pub pubkey: PublicKey,
+    pub connection_string: String,
+    pub url: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct GetInfoResponse {
+    pub pubkey: String,
+    pub connection_methods: Vec<GetinfoAddress>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub(crate) struct GetinfoAddress {
+    #[serde(rename = "type")]
+    pub item_type: GetinfoAddressType,
+    #[serde(alias = "port")]
+    pub port: u16,
+    #[serde(alias = "address", skip_serializing_if = "Option::is_none")]
+    pub address: Option<String>,
+}
+
+/// Type of connection
+#[derive(Copy, Clone, Debug, Deserialize, Serialize)]
+pub(crate) enum GetinfoAddressType {
+    #[serde(rename = "dns")]
+    DNS,
+    #[serde(rename = "ipv4")]
+    IPV4,
+    #[serde(rename = "ipv6")]
+    IPV6,
+    #[serde(rename = "torv2")]
+    TORV2,
+    #[serde(rename = "torv3")]
+    TORV3,
+    #[serde(rename = "websocket")]
+    WEBSOCKET,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ProposalRequest {
+    pub bolt11: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub host: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub port: Option<u16>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ProposalResponse {
+    pub jit_bolt11: String,
+}
+
+const GET_INFO_PATH: &str = "/api/v1/info";
+const PROPOSAL_PATH: &str = "/api/v1/proposal";
+
+impl LspClient {
+    pub async fn new(url: String) -> anyhow::Result<Self> {
+        let http_client = Client::builder().build()?;
```

Comment:
you could add the http_client to `LspClient` that way we dont need to reinitialize it every call